File: code/JL_OTA_SDK/otasdk_java/src/main/java/com/jieli/otasdk_java/tool/IOtaContract.java
package com.jieli.otasdk_java.tool;

import android.bluetooth.BluetoothDevice;

import com.jieli.otasdk_java.base.BasePresenter;
import com.jieli.otasdk_java.base.BaseView;
import com.jieli.otasdk_java.tool.ota.OTAManager;

/**
 * @ClassName: IOtaContract
 * @Description: java类作用描述
 * @Author: ZhangHuanMing
 * @CreateDate: 2021/12/23 8:51
 */
public interface IOtaContract {
    public interface IOtaPresenter extends BasePresenter {
        public BluetoothDevice getConnectedDevice();

        public boolean isDevConnected();

        public OTAManager getOtaManager();

        public boolean isOTA();

        public void startOTA(String filePath);

        public void cancelOTA();

        public void reconnectDev(String devAddr);
    }

    public interface IOtaView extends BaseView<IOtaPresenter> {
        public boolean isViewShow();

        public void onConnection(BluetoothDevice device, int status);

        public void onMandatoryUpgrade();

        public void onOTAStart();

        public void onOTAReconnect(String btAddr);

        public void onOTAProgress(int type, float progress);

        public void onOTAStop();

        public void onOTACancel();

        public void onOTAError(int code, String message);
    }
}

File: code/JL_OTA_SDK/otasdk/src/main/java/com/jieli/otasdk/tool/IOtaContract.kt
package com.jieli.otasdk.tool

import android.bluetooth.BluetoothDevice
import com.jieli.otasdk.base.BasePresenter
import com.jieli.otasdk.base.BaseView
import com.jieli.otasdk.tool.ota.OTAManager

/**
 * OTA接口管理
 *
 * @author zqjasonZhong
 * @date 2019/12/30
 */
interface IOtaContract {

    interface IOtaPresenter : BasePresenter {

        fun getConnectedDevice(): BluetoothDevice?

        fun isDevConnected(): Boolean

        fun getOtaManager(): OTAManager?

        fun isOTA(): Boolean

        fun startOTA(filePath: String)

        fun cancelOTA()

        fun reconnectDev(devAddr: String?)
    }

    interface IOtaView : BaseView<IOtaPresenter> {
        fun isViewShow(): Boolean

        fun onConnection(device: BluetoothDevice?, status: Int)

        fun onMandatoryUpgrade()

        fun onOTAStart()

        fun onOTAReconnect(btAddr: String?)

        fun onOTAProgress(type: Int, progress: Float)

        fun onOTAStop()

        fun onOTACancel()

        fun onOTAError(code: Int, message: String)
    }
}